app/routes/messages.routes.js

```diff
@@ -0,0 +1,18 @@
+var messages = require('../controllers/messages.controller');
+var auth = require('../utils/auth-middleware.utils');
+
+module.exports = function (app) {
+    app.get('/messages/:listing_id/:first_user_id/:second_user_id',
```

Comment:
I don't think you actually need to specify both users for this. Since the logged in user will always be one either the sender or receiver of an message, you can always get one of the users from the token. Also, to satisfy REST convention, for path parameters, each parameter needs it's own path. For example, this endpoint would be:
`/messages/:listingId/first-user/:firstUserId/second-user/:secondUserId`

Although as I stated above, I believe you can get away with:
`/messages/:listingId/users/:userId`

Reply:
Also, if you are the owner of a Listing, it might also be useful to have an endpoint that gives you all of the messages for that listing:
`/messages/:listingId`